### PR TITLE
Restore warn(missing_docs) in the Exonum crate [ECR-3821]

### PR DESCRIPTION
**Pull request by alekseysidorov**

### CODE REVIEW

File: exonum/src/blockchain/schema.rs [hunk 1]
```diff
@@ -324,6 +324,7 @@ where
     }
 }
 
+/// Describes affiliation of the index.
```
Comment (popzxc) [inline exonum/src/blockchain/schema.rs]: Not sure if "affiliation" is applicable here.

File: exonum/src/proto/mod.rs [hunk 2]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Module for all protobuf messages used by Exonum.
```
Comment (popzxc) [inline exonum/src/proto/mod.rs]:
```suggestion
//! Module containing protobuf messages used by Exonum.
```

File: exonum/src/api/mod.rs [hunk 3]
```diff
@@ -218,10 +218,12 @@ impl ApiAggregator {
         Self { endpoints }
     }
 
+    /// Inserts an endpoints handler with the given mount point.
     pub fn insert(&mut self, name: &str, api: ApiBuilder) {
         self.endpoints.insert(name.to_owned(), api);
     }
 
+    /// Extends the list of endpoint handlers with the specified ones.
```
Comment (Maria-Nosyk) [inline exonum/src/api/mod.rs]:
```suggestion
    /// Extends the list of endpoint handlers with the new specified handlers.
```

File: exonum/src/blockchain/builder.rs [hunk 4]
```diff
@@ -68,6 +69,8 @@ impl BlockchainBuilder {
         self.with_additional_runtime(runtime)
     }
 
+    /// Adds a multiple runtimes with the specified identifiers and returns
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/builder.rs]:
```suggestion
    /// Adds multiple runtimes with the specified identifiers and returns
```

File: exonum/src/blockchain/builder.rs [hunk 5]
```diff
@@ -78,13 +81,14 @@ impl BlockchainBuilder {
...
-    /// Add an additional runtime with the specified identifier.
+    /// Adds an additional runtime with the specified identifier and returns
+    /// a modified `Self` object for further chaining.
     pub fn with_additional_runtime(mut self, runtime: impl Into<(u32, Box<dyn Runtime>)>) -> Self {
         self.runtimes.push(runtime.into());
         self
     }
 
-    /// Add instance specifications of builtin services.
+    /// Adds instance specifications of builtin services.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/builder.rs]:
```suggestion
    /// Adds instance specifications of the built-in services.
```

File: exonum/src/blockchain/builder.rs [hunk 6]
```diff
@@ -134,6 +138,7 @@ pub struct InstanceConfig {
 }
 
 impl InstanceConfig {
+    /// Creates a new instantiation parameters of service instance.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/builder.rs]:
```suggestion
    /// Creates new instantiation parameters of the service instance.
```

File: exonum/src/blockchain/schema.rs [hunk 7]
```diff
@@ -184,20 +184,17 @@ where
         Entry::new(CONSENSUS_CONFIG, self.access.clone())
     }
 
-    /// Returns the accessory `ProofMapIndex` for calculating
-    /// patches in the DBView layer.
+    // TODO Write more meaningful description [ECR-3824]
+    /// Returns the accessory `ProofMapIndex` for calculating patches.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
    /// Returns an accessory `ProofMapIndex` for calculating patches.
```

File: exonum/src/blockchain/schema.rs [hunk 8]
```diff
@@ -184,20 +184,17 @@ where
...
     ///
     /// The table calculates the "aggregation" of root hashes of individual
     /// service tables, in effect summing the state of various entities,
     /// scattered across distinct services and their tables. Sum is performed by
     /// means of computing the root hash of this table.
     ///
-    /// - Table **key** is  normalized coordinates of a service.
-    /// - Table **value** is the root hash of a service table, which contributes
+    /// - Table **key** is normalized coordinates of an index.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
    /// - Table **key** contains normalized coordinates of an index.
```

File: exonum/src/blockchain/schema.rs [hunk 9]
```diff
@@ -184,20 +184,17 @@ where
...
+    /// - Table **value** is the root hash of an index, which contributes
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
    /// - Table **value** contains a root hash of the index, which contributes
```

File: exonum/src/blockchain/schema.rs [hunk 10]
```diff
@@ -324,13 +321,15 @@ where
     }
 }
 
+// TODO Write more meaningful description [ECR-3824]
+/// Describes the membership for the index.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum IndexOwner {
     /// This index is part of core schema.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
    /// This index is a part of the core schema.
```

File: exonum/src/blockchain/schema.rs [hunk 11]
```diff
@@ -324,13 +321,15 @@ where
...
     Core,
-    /// This index is a part of runtime schema.
+    /// This index is a part of runtime schema with the specified ID.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
    /// This index is a part of the runtime schema with the specified ID.
```

File: exonum/src/blockchain/schema.rs [hunk 12]
```diff
@@ -368,6 +367,8 @@ enum IndexTag {
     Service = 3,
 }
 
+// TODO Write more meaningful description [ECR-3824]
+/// Normalized coordinates of index in the `state_hash_aggregator` table.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
/// Normalized coordinates of the index in the `state_hash_aggregator` table.
```

File: exonum/src/blockchain/schema.rs [hunk 13]
```diff
@@ -385,6 +386,9 @@ impl IndexCoordinates {
         }
     }
 
+    // TODO Write more meaningful description [ECR-3824]
+    /// For a given index owner, returns a list if index coordinates for the corresponding
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
    /// For the given index owner, returns a list of the index coordinates that match the corresponding
```

File: exonum/src/blockchain/schema.rs [hunk 14]
```diff
@@ -385,6 +386,9 @@ impl IndexCoordinates {
...
+    /// object hashes of the indexes.
```
Comment (Maria-Nosyk) [inline exonum/src/blockchain/schema.rs]:
```suggestion
    /// object hashes of the indices.
```

File: exonum/src/node/mod.rs [hunk 15]
```diff
@@ -302,6 +302,7 @@ impl NodeConfig {
         }
     }
 
+    /// Returns the service keypair of the node.
```
Comment (Maria-Nosyk) [inline exonum/src/node/mod.rs]:
```suggestion
    /// Returns a service key pair of the node.
```

File: exonum/src/node/mod.rs [hunk 16]
```diff
@@ -958,6 +959,7 @@ impl Node {
         Self::with_blockchain(blockchain, channel, node_cfg, config_file_path)
     }
 
+    /// Creates node for the given blockchain and node configuration.
```
Comment (Maria-Nosyk) [inline exonum/src/node/mod.rs]:
```suggestion
    /// Creates a node for the given blockchain and node configuration.
```

File: exonum/src/proto/mod.rs [hunk 17]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Module containing protobuf messages used by Exonum.
```
Comment (Maria-Nosyk) [inline exonum/src/proto/mod.rs]:
```suggestion
//! Module that contains Protobuf messages used by Exonum.
```

File: exonum/src/runtime/rust/call_context.rs [hunk 18]
```diff
@@ -43,6 +43,7 @@ impl<'a> CallContext<'a> {
         Schema::new(self.fork())
     }
 
+    /// Returns the descriptor of the executing service instance.
```
Comment (Maria-Nosyk) [inline exonum/src/runtime/rust/call_context.rs]:
```suggestion
    /// Returns a descriptor of the executing service instance.
```

File: exonum/src/runtime/types.rs [hunk 19]
```diff
@@ -326,7 +326,9 @@ impl Display for InstanceSpec {
 /// Allows to query a service instance by either of the two identifiers.
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub enum InstanceQuery<'a> {
+    /// Query by instance ID.
```
Comment (Maria-Nosyk) [inline exonum/src/runtime/types.rs]:
```suggestion
    /// Query by an instance ID.
```

File: exonum/src/runtime/types.rs [hunk 20]
```diff
@@ -326,7 +326,9 @@ impl Display for InstanceSpec {
...
     Id(InstanceId),
+    /// Query by instance name.
```
Comment (Maria-Nosyk) [inline exonum/src/runtime/types.rs]:
```suggestion
    /// Query by an instance name.
```

**Pull request merged**